src/Compiler/Checking/NicePrint.fs

```diff
@@ -2245,8 +2246,39 @@ module InferredSigPrinting =
             let innerPath = (fullCompPathOfModuleOrNamespace mspec).AccessPath
             let outerPath = mspec.CompilationPath.AccessPath
 
-            let denv = denv.AddOpenPath (List.map fst innerPath) 
-            if mspec.IsNamespace then
+            let denv = denv.AddOpenPath (List.map fst innerPath)
+
+            let isWrappedNamespace =
+                match mspec.ModuleOrNamespaceType.ModuleOrNamespaceKind with
+                | Namespace false -> true
+                | _ -> false
+
+            if isWrappedNamespace then
+                // The current mspec is a namespace that belongs to the `def` child (nested) module(s).                
+                let fullModuleName, def =
+                    let rec (|NestedModule|_|) (currentContents:ModuleOrNamespaceContents) =
+                        match currentContents with
+                        | ModuleOrNamespaceContents.TMDefRec (bindings = [ ModuleOrNamespaceBinding.Module(mn, NestedModule(path, contents)) ]) ->
+                            Some ([ yield mn.DisplayNameCore; yield! path ], contents)
+                        | ModuleOrNamespaceContents.TMDefs [ ModuleOrNamespaceContents.TMDefRec (bindings = [ ModuleOrNamespaceBinding.Module(mn, NestedModule(path, contents)) ]) ] ->
```

Comment:
I am wondering if this can end up stripping off too many

Reply:
@dsyme this would create:

```fsharp
module A.B.C

val x: int
```

Which doesn't seem that bad to me.

If there are more things in the top-level module, it will not absorb the nested module.
Example:
```fsharp
module A.B

module C =
   let x = 1

let y = 2
```

will lead to

```fsharp
module A.B

module C =
  
  val x: int

val y: int
```

Reply:
OK thanks. I guess it's OK! :)